src/filters/blur/BlurDirFilter.js

```diff
@@ -0,0 +1,143 @@
+var core = require('../../core');
+var fs = require('fs');
+
+/**
+ * The BlurDirFilter applies a Gaussian blur toward a direction to an object.
+ *
+ * @class
+ * @param {number} dirX
+ * @param {number} dirY
+ * @extends AbstractFilter
+ * @memberof PIXI.filters
+ */
+function BlurDirFilter(dirX, dirY)
+{
+    core.AbstractFilter.call(this,
+        // vertex shader
+        fs.readFileSync(__dirname + '/blurDir.vert', 'utf8'),
+        // fragment shader
+        fs.readFileSync(__dirname + '/blurDir.frag', 'utf8'),
+        // set the uniforms
+        {
+            strength: { type: '1f', value: 1 },
+            dirX: { type: '1f', value: dirX || 0 },
+            dirY: { type: '1f', value: dirY || 0 }
+        }
+    );
+
+    this.defaultFilter = new core.AbstractFilter();
+
+    /**
+     * Sets the number of passes for blur. More passes means higher quaility bluring.
+     *
+     * @member {number}
+     * @memberof BlurDirFilter#
+     * @default 1
+     */
+    this.passes = 1;
+
+    /**
+     * Sets the X direction of the blur
+     *
+     * @member {number}
+     * @memberof BlurDirFilter#
+     * @default 0
+     */
+    this.dirX = dirX || 0;
+
+    /**
+     * Sets the Y direction of the blur
+     *
+     * @member {number}
+     * @memberof BlurDirFilter#
+     * @default 0
+     */
+    this.dirY = dirY || 0;
+
+    this.strength = 4;
+}
+
+BlurDirFilter.prototype = Object.create(core.AbstractFilter.prototype);
+BlurDirFilter.prototype.constructor = BlurDirFilter;
+module.exports = BlurDirFilter;
+
+BlurDirFilter.prototype.applyFilter = function (renderer, input, output, clear) {
+
+    var shader = this.getShader(renderer);
+
+    this.uniforms.strength.value = this.strength / 4 / this.passes * (input.frame.width / input.size.width);
+
+    if (this.passes === 1) {
```

Comment:
Please fix the braces in this block to use the newline-first style used throughout the code base :)